### PR TITLE
feat(es): support TypeScript Instantiation Expression

### PR DESCRIPTION
**Pull request by g-plane**
**Description:**

This PR added support of TypeScript new feature: [Instantiation Expression](https://github.com/microsoft/TypeScript/pull/47607).

### CODE REVIEW

File: crates/swc_ecma_ast/src/expr.rs [hunk 1]
```diff
@@ -147,6 +147,9 @@ pub enum Expr {
     #[tag("TsAsExpression")]
     TsAs(TsAsExpr),
 
+    #[tag("TsExpressionWithTypeArguments")]
+    TsExprWithTypeArgs(TsExprWithTypeArgs),
```
Comment (kdy1) [inline crates/swc_ecma_ast/src/expr.rs:151]: Is this required?
Also, it does not seem like a good name.

Comment (g-plane) [reply crates/swc_ecma_ast/src/expr.rs:151]: I think it's required. The name is borrowed from TypeScript and shortened. Or, do you have any suggestions?

Comment (kdy1) [reply crates/swc_ecma_ast/src/expr.rs:151]: `Expr:: TsExprWithTypeArgs` does not seems like an instantiation expression. How about `TsInstantiation`?

Comment (g-plane) [reply crates/swc_ecma_ast/src/expr.rs:151]: LGTM.

**Review by kdy1 — COMMENTED** [review]
LGTM. Thank you!

---

swc-bump:
 - swc_ecma_ast --breaking

**Review by kdy1 — APPROVED** [review]

**Pull request merged**